### ISSUE TITLE
End-to-end Learning of Deep Visual Representations for Image Retrieval

### ISSUE THREAD

**Issue by chullhwan-song** [chullhwan-song/Reading-Paper#17]
https://arxiv.org/abs/1610.07940

**Comment by chullhwan-song** [comment]
* what ?
   * instance-level image retrieval
   * r-mac descriptor - learnable
   * landmark dataset - https://github.com/chullhwan-song/Reading-Paper/issues/14 연구에서 소개한 neuralcode dataset를 이용. 이 데이터는 언급한거와 같이 noisy한 데이터 -> clean 
   * triplet loss 기반 학습. > 참고 https://github.com/chullhwan-song/Reading-Paper/issues/16
   * Oxford 5k, Paris 6k and Holidays testset 적용 - delf https://github.com/chullhwan-song/Reading-Paper/issues/4 연구가 나올때까지 SOTA 성능
   * 구 제록스 연구소 현 네이버랩스유럽 소속 논문
* 학습셋 구성
   * neuralcode dataset  noisy한 데이터 -> clean 를 clean 데이터로 구성 
![image](https://user-images.githubusercontent.com/40360823/42607017-aada60da-85bb-11e8-9c2f-8c3ee1f51aad.png)
       * SIFT & Hessian-Ane keypoint 이용 -> bow, kNN -> graph : 32-core server
       * This cleaning process leaves about 49,000 images (divided in 42,410 training and 6,382 validation images) still belonging to one of the 586 landmarks,
       * [링크](http://www.europe.naverlabs.com/Research/Computer-Vision/Learning-Visual-Representations/Deep-Image-Retrieval) 다운로드 가능 -> delf에서도 이를 이용.
     * Bounding box estimation 부분이 있는데, 이는 R-MAC의 local region을 획득하는 방법(원래는 rigid grid)  
![image](https://user-images.githubusercontent.com/40360823/42607286-0ebb33d0-85bd-11e8-9016-67d89f8ac0c7.png)
* [R-MAC](https://arxiv.org/abs/1511.05879) 
   * 이 논문의 기본 descriptor 알고리즘은 MAC 인데 이를 local 특성을 가진 feature 업그레이드 한것이 R-MAC
   * 현재 많이 적용/응용되어 사용하고 있는 인기있는 feature
   * MAC은 conv feature map으로 부터 channel dimension의 feature 를 생성, 한 채널은 wxh의 크기를 가지는데 이때 wxh의 max값을 취함.
![image](https://user-images.githubusercontent.com/40360823/42607363-6a7bfbdc-85bd-11e8-84bf-cd4a3c65a50f.png)
   * 이 방식을 기반으로 하고, pre-processing단계에서, 다음과 같이 rigid grid region 획득
![image](https://user-images.githubusercontent.com/40360823/42607390-8b46fad8-85bd-11e8-91e8-c21012dd2b42.png)
   * 이 region들을 기반(conv feature map기준으로 보면 sub region)으로 MAC feature를 취함.
   * 이후에 각 region별로, l2 norm -> whitening -> l2 하고 각 region별로 sum을 한후, 다시 l2 norm을 하여 최종적으로 feature를 취함
      * whitening은 base는 PCA 인데, 원 MAC 들은 learnable feature가 아님. 그래서, 이를 해결하기 위해 , shift+fc, 즉, affine transformation을 하는것 같음. (backpropa에서 미분가능하도록 만들어줘야함)
![image](https://user-images.githubusercontent.com/40360823/42607488-25c64000-85be-11e8-9974-b2a27bb4c47d.png)
    * 또한 앞에서 언급한 Bounding box estimation 에서 획득한 region들은 위의 rigid grid 방식을 대체함 
![image](https://user-images.githubusercontent.com/40360823/42607539-6f5d0028-85be-11e8-8500-7c8027c47aef.png)
* triplet loss 기반 학습
![image](https://user-images.githubusercontent.com/40360823/42607717-4913e174-85bf-11e8-9ded-70b6c46e2fab.png)
   * 최종적으로 learnable 형태로 만들고 학습을 진행 
![image](https://user-images.githubusercontent.com/40360823/42607579-9ccb0280-85be-11e8-9806-a04250855ee8.png)
   * classification loss와의 비교
   * triplet loss 기반 학습은 기본적으로 negative sample을 획득하는 방법이 중요.(facenet 참고바람)
   * 자세히 쓰진 않겠지만, 논문상에선, vgg/resnet101같은 network을 이용. 하지만 gpu 메모리등 이유로 어려움 겪음... >해결방법제시. 
* 실험
![image](https://user-images.githubusercontent.com/40360823/42607743-60f81846-85bf-11e8-9802-fda563ddfe3c.png)
   * 결과적으로 delf나오기 전까지 SOTA(너무 올려놨음 ㅠ)
   * 이외 다양한 실험했는데, resnet101사용하여 SOTA
   * classification 보다 triplet loss가 더 좋다.
   * rigid grid는 성능이 낮으나 그리 차이가 나지 않는다. 
   * clean data로 하니 더 좋다. fine-tuning
   * 이외 QE, multi-resolution,  compression했는데 PQ방식이 젤 좋다 등등
